Source/AudioPlayer.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  AudioPlayer.swift
+//  Pods-VEAudioKitDemo
+//
+//  Created by Pablo Balduz on 10/07/2019.
+//
+
+import AVFoundation
+
+public protocol AudioPlayerDelegate {
+    func playerDidEnd()
+    func playerDidUpdatePosition(seconds: Float)
+}
+
+public class AudioPlayer {
+    
+    private var engine = AVAudioEngine()
+    private var player = AVAudioPlayerNode()
+    private var scheduler: Scheduler!
+    
+    private var audioFile: AVAudioFile? {
+        didSet {
+            if let audioFile = audioFile {
+                audioLengthSamples = audioFile.length
+                audioFormat = audioFile.processingFormat
+                audioSampleRate = Float(audioFormat?.sampleRate ?? 44100)
+                audioLengthSeconds = Float(audioLengthSamples) / audioSampleRate
+            }
+        }
+    }
+    
+    private var audioFormat: AVAudioFormat?
+    private var audioSampleRate: Float = 0
+    private var audioLengthSeconds: Float = 0
+    private var audioLengthSamples: AVAudioFramePosition = 0
+    
+    private var timer: Timer?
+    private var currentFrame: AVAudioFramePosition {
+        guard let lastRenderTime = player.lastRenderTime, let playerTime = player.playerTime(forNodeTime: lastRenderTime) else {
+            return 0
+        }
+        return playerTime.sampleTime
+    }
+    private var currentPosition: AVAudioFramePosition = 0
+    
+    public var delegate: AudioPlayerDelegate?
+    
+    public var audioFileURL: URL? {
+        didSet {
+            if let url = audioFileURL {
+                audioFile = try? AVAudioFile(forReading: url)
+            }
+        }
+    }
+    
+    public var isPlaying: Bool {
+        return player.isPlaying
+    }
+    
+    public var duration: Float {
+        return audioLengthSeconds
+    }
+    
+    public init() { }
+    
+    public func prepare() {
```

Comment:
It's necessary to prepare after adding any new file to the player. Maybe it can be called during init with an initial audio file as parameter and also automatically called after appending new audio files to the audio files array. Should that be a better option?

Reply:
It's necessary to prepare after adding any new file to the player. Maybe it can be called during init with an initial audio file as parameter and also automatically called after appending new audio files to the audio files array. Should that be a better option?